### PR TITLE
helper: request another PD if one of them is unavailable

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/35708.

Problem Summary:

`requestPD` should try the PD hosts one by one until the request is successful.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix an issue that TiDB cannot access `TIKV_REGION_STATUS` when one of the PD is down.
```

### CODE REVIEW

File: store/helper/helper.go [hunk 1]
```diff
@@ -840,20 +840,21 @@ func (h *Helper) requestPD(apiName, method, uri string, body io.Reader, res inte
...
-			// Try to request from another PD node when some nodes may down.
-			if strings.Contains(err.Error(), "connection refused") {
-				continue
-			}
-			return errors.Trace(err)
+		err = requestPDForOneHost(host, apiName, method, uri, body, res)
+		if err == nil {
+			break
 		}
+		// Try to request from another PD node when some nodes may down.
```
Comment (JmPotato) [inline store/helper/helper.go:848]: Should we print a log here to show the error? Otherwise, only the last host's error will be returned.

Comment (tangenta) [reply store/helper/helper.go:848]: @JmPotato Updated. The errors are logged now.

File: store/helper/helper.go [hunk 2]
```diff
@@ -851,23 +851,29 @@ func (h *Helper) requestPD(apiName, method, uri string, body io.Reader, res inte
 }
 
 func requestPDForOneHost(host, apiName, method, uri string, body io.Reader, res interface{}) error {
-	logutil.BgLogger().Debug("RequestPD URL", zap.String("url", util.InternalHTTPSchema()+"://"+host+uri))
-	req, err := http.NewRequest(method, util.InternalHTTPSchema()+"://"+host+uri, body)
+	urlVar := fmt.Sprintf("%s://%s%s", util.InternalHTTPSchema(), host, uri)
+	logutil.BgLogger().Debug("RequestPD URL", zap.String("url", urlVar))
+	req, err := http.NewRequest(method, urlVar, body)
 	if err != nil {
+		logutil.BgLogger().Warn("requestPDForOneHost new request",
```
Comment (XuHuaiyu) [inline store/helper/helper.go:858]: requestPDForOneHost new request fail

File: store/helper/helper.go [hunk 3]
```diff
@@ -851,23 +851,29 @@ func (h *Helper) requestPD(apiName, method, uri string, body io.Reader, res inte
...
+			zap.String("url", urlVar), zap.Error(err))
 		return errors.Trace(err)
 	}
 	start := time.Now()
 	resp, err := util.InternalHTTPClient().Do(req)
 	if err != nil {
 		metrics.PDAPIRequestCounter.WithLabelValues(apiName, "network error").Inc()
+		logutil.BgLogger().Warn("requestPDForOneHost do request",
```
Comment (XuHuaiyu) [inline store/helper/helper.go:866]: requestPDForOneHost do request fail

File: store/helper/helper.go [hunk 4]
```diff
@@ -851,23 +851,29 @@ func (h *Helper) requestPD(apiName, method, uri string, body io.Reader, res inte
...
+			zap.String("url", urlVar), zap.Error(err))
 		return errors.Trace(err)
 	}
 	metrics.PDAPIExecutionHistogram.WithLabelValues(apiName).Observe(time.Since(start).Seconds())
 	metrics.PDAPIRequestCounter.WithLabelValues(apiName, resp.Status).Inc()
 	defer func() {
 		err = resp.Body.Close()
 		if err != nil {
-			logutil.BgLogger().Error("close body failed", zap.Error(err))
+			logutil.BgLogger().Warn("requestPDForOneHost close body",
```
Comment (XuHuaiyu) [inline store/helper/helper.go:875]: requestPDForOneHost close body fail

**Review by XuHuaiyu — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Pull request merged**